pkg/cmd/roachprod/main.go

```diff
@@ -997,21 +997,21 @@ of nodes, outputting a line whenever a change is detected:
 		if err != nil {
 			return err
 		}
-		var lastErr error
+		var errs []string
 		for msg := range c.Monitor(monitorIgnoreEmptyNodes, monitorOneShot) {
 			if msg.Err != nil {
-				lastErr = errors.Wrapf(err, "%d", msg.Index)
-				fmt.Printf("%d: error: %s\n", msg.Index, msg.Err)
-				continue
-			}
-			if msg.Msg != "" {
-				fmt.Printf("%d: %s\n", msg.Index, msg.Msg)
+				msg.Msg += "error: " + msg.Err.Error()
 			}
-			if strings.Contains(msg.Msg, "dead") {
-				lastErr = errors.Wrapf(errors.New(msg.Msg), "%d", msg.Index)
+			s := fmt.Sprintf("%d: %s", msg.Index, msg.Msg)
+			if msg.Err != nil || strings.Contains(msg.Msg, "dead") {
+				errs = append(errs, s)
 			}
+			fmt.Println(s)
 		}
-		return lastErr
+		if len(errs) != 0 {
+			return errors.New(strings.Join(errs, ", "))
```

Comment:
Would using a newline separator be more readable? I guess I could use an example of what this looks like, though I trust you’ve considered that.

Reply:
1: dead, 2: dead

This is typically what you see.